Issue2884 pid autotuning

SenHuang19 opened the pull request:
This is to address the comments in #2957
@JayHuLBL Can you please review it?

JayHuLBL commented:
@SenHuang19 
By running `../bin/verifyFiles.py`, I see following errors:
```
jianjunhu@ubuntu:~/GitFolder/SenHuang19/modelica-buildings/Buildings$ ../bin/verifyFiles.py 
*** Error:  File 'Controls/OBC/Utilities/PIDWithAutotuning/Relay/BaseClasses/package.mo' does not contain required string 'documentation'.
*** Error:  File 'Controls/OBC/Utilities/PIDWithAutotuning/SystemIdentification/FirstOrderTimedelayed/BaseClasses/package.mo' does not contain required string 'documentation'.
*** Error:  File 'Controls/OBC/Utilities/PIDWithAutotuning/AutoTuner/AMIGO/BaseClasses/package.mo' does not contain required string 'documentation'.
*** Terminating due to found errors in examined files.
```

JayHuLBL commented on Buildings/Controls/OBC/Utilities/PIDWithAutotuning/AutoTuner/AMIGO/BaseClasses/Validation/PIDGain.mo:
```diff
@@ -0,0 +1,70 @@
+within Buildings.Controls.OBC.Utilities.PIDWithAutotuning.AutoTuner.AMIGO.BaseClasses.Validation;
+model PIDGain "Test model for calculating the control gain for a PID controller"
+  Buildings.Controls.OBC.Utilities.PIDWithAutotuning.AutoTuner.AMIGO.BaseClasses.PIDGain
+                                                                             PIDGai
```
change to one line.

SenHuang19 replied:
Fix the format.

JayHuLBL commented on Buildings/Controls/OBC/Utilities/PIDWithAutotuning/AutoTuner/AMIGO/BaseClasses/Validation/PIGain.mo:
```diff
@@ -0,0 +1,70 @@
+within Buildings.Controls.OBC.Utilities.PIDWithAutotuning.AutoTuner.AMIGO.BaseClasses.Validation;
+model PIGain "Test model for calculating the gain for a PI controller"
+  Buildings.Controls.OBC.Utilities.PIDWithAutotuning.AutoTuner.AMIGO.BaseClasses.PIGain
+                                                                            PIGai
```
one line.

JayHuLBL commented on Buildings/Controls/OBC/Utilities/PIDWithAutotuning/AutoTuner/AMIGO/BaseClasses/package.mo:
```diff
@@ -0,0 +1,16 @@
+          lineColor={128,128,128},
+          fillColor={255,255,255},
+          fillPattern=FillPattern.Solid)}));
+end BaseClasses;
```
missing `Documentation` section.

SenHuang19 replied:
Add <code>Documentation</code>.

JayHuLBL commented on Buildings/Controls/OBC/Utilities/PIDWithAutotuning/AutoTuner/AMIGO/PI.mo:
```diff
@@ -30,11 +30,12 @@ block PI "Identify control gain and integral time of the PI controller"
-  Buildings.Controls.OBC.Utilities.PIDWithAutotuning.AutoTuner.AMIGO.PIGain gai
-    "Calculate the control gain"
+protected
+  BaseClasses.PIGain                                                        gai
```
use full path, remove the empty spaces.

JayHuLBL commented on Buildings/Controls/OBC/Utilities/PIDWithAutotuning/AutoTuner/AMIGO/PI.mo:
```diff
@@ -30,11 +30,12 @@ block PI "Identify control gain and integral time of the PI controller"
     annotation (Placement(transformation(extent={{-10,20},{10,40}})));
-  Buildings.Controls.OBC.Utilities.PIDWithAutotuning.AutoTuner.AMIGO.PIIntegralTime
-    intTim "Calculate the integral time"
+  BaseClasses.PIIntegralTime
```
full path, one line.

SenHuang19 replied:
The full path is used and the format is fixed

JayHuLBL commented on Buildings/Controls/OBC/Utilities/PIDWithAutotuning/AutoTuner/AMIGO/PI.mo:
```diff
@@ -62,7 +63,40 @@ equation
-          textColor={0,0,255})}), Diagram(coordinateSystem(preserveAspectRatio=false)),
+          textColor={0,0,255}),
+        Text(
+          visible=(controllerType == Buildings.Controls.OBC.CDL.Types.SimpleController.PI),
```
I see this line in various models. It seems should not be here (and the model does not have parameter `controllerType`).
Please check other models as well.

SenHuang19 replied:
Remove this line from this model and other models.

JayHuLBL commented on Buildings/Controls/OBC/Utilities/PIDWithAutotuning/AutoTuner/AMIGO/PID.mo:
```diff
@@ -27,23 +27,24 @@ block PID "Identify control gain, integral time, and derivative time of the PID
-  Buildings.Controls.OBC.Utilities.PIDWithAutotuning.AutoTuner.AMIGO.PIDGain gai
-    "Calculate the control gain"
+protected
+  BaseClasses.PIDGain                                                        gai
```
full path, one line.

SenHuang19 replied:
The full path is used and the format is fixed.

JayHuLBL commented on Buildings/Controls/OBC/Utilities/PIDWithAutotuning/Relay/ResponseProcess.mo:
```diff
@@ -37,19 +37,20 @@ block ResponseProcess
     annotation (Placement(transformation(extent={{-80,0},{-60,20}})));
-  Buildings.Controls.OBC.Utilities.PIDWithAutotuning.Relay.HalfPeriodRatio halPerRatio
-    "Calculate the half period ratio"
+  BaseClasses.HalfPeriodRatio                                              halPerRatio
```
full path, one line.

SenHuang19 replied:
The full path is used and the format is fixed

JayHuLBL reviewed (changes requested):
@SenHuang19 Please see inline comments.

SenHuang19 commented:
> @SenHuang19
> 
> * By running `../bin/verifyFiles.py`, I see following errors:
> 
> ```
> jianjunhu@ubuntu:~/GitFolder/SenHuang19/modelica-buildings/Buildings$ ../bin/verifyFiles.py 
> *** Error:  File 'Controls/OBC/Utilities/PIDWithAutotuning/Relay/BaseClasses/package.mo' does not contain required string 'documentation'.
> *** Error:  File 'Controls/OBC/Utilities/PIDWithAutotuning/SystemIdentification/FirstOrderTimedelayed/BaseClasses/package.mo' does not contain required string 'documentation'.
> *** Error:  File 'Controls/OBC/Utilities/PIDWithAutotuning/AutoTuner/AMIGO/BaseClasses/package.mo' does not contain required string 'documentation'.
> *** Terminating due to found errors in examined files.
> ```
> 
> * Also, following hyperlinks are broken:
> 
> ```
> Error: Can not resolve link "modelica://Buildings.Controls.OBC.Utilities.PIDWithAutotuning.Relay.HalfPeriodRatio" in Buildings.Controls.OBC.Utilities.PIDWithAutotuning.FirstOrderAMIGO
> Error: Can not resolve link "modelica://Buildings.Controls.OBC.Utilities.PIDWithAutotuning.Relay.Validation.NormalizedTimeDelay" in Buildings.Controls.OBC.Utilities.PIDWithAutotuning.Relay.BaseClasses.NormalizedTimeDelay
> Error: Can not resolve link "modelica://Buildings.Controls.OBC.Utilities.PIDWithAutotuning.SystemIdentification.FirstOrderTimedelayed.Validation.TimeConstantDelay" in Buildings.Controls.OBC.Utilities.PIDWithAutotuning.SystemIdentification.FirstOrderTimedelayed.BaseClasses.TimeConstantDelay
> Error: Can not resolve link "modelica://Buildings.Controls.OBC.Utilities.PIDWithAutotuning.SystemIdentification.FirstOrderTimedelayed.BaseClassess.Gain" in Buildings.Controls.OBC.Utilities.PIDWithAutotuning.SystemIdentification.FirstOrderTimedelayed.BaseClasses.Validation.Gain
> Error: Can not resolve link "modelica://Buildings.Controls.OBC.Utilities.PIDWithAutotuning.SystemIdentification.FirstOrderTimedelayed.BaseClassess.TimeConstantDelay" in Buildings.Controls.OBC.Utilities.PIDWithAutotuning.SystemIdentification.FirstOrderTimedelayed.BaseClasses.Validation.TimeConstantDelay
> Generated documentation for Buildings.Controls.OBC.Utilities.PIDWithAutotuning in /home/jianjunhu/GitFolder/SenHuang19/modelica-buildings/Buildings/help/Buildings_Controls_OBC_Utilities_PIDWithAutotuning.html
> ```
> 
> * in `Buildings.Controls.OBC.Utilities.PIDWithAutotuning.FirstOrderAMIGO`, for the Icon layer text string "PI", change the `textString="PI"` to `textString= if controllerType == Buildings.Controls.OBC.CDL.Types.SimpleController.PID then "PID" else "PI"` and remove the `visible=(controllerType == Buildings.Controls.OBC.CDL.Types.SimpleController.PI),`.
> * please also see the inline comments below.

- The document is added
- The broken links are fixed
- The icon is updated

SenHuang19 commented:
> @SenHuang19 Please see inline comments.
@JayHuLBL thanks a lot for the comments. 
Most of them are due to the recent restructure of the sub-packages.
I believe I addressed all of your comments.
Can you please review it again?
